doc: Add instructions to access cluster with IBM Cloud vpc-gen2. (#2530)
kubeflow/website: content/en/docs/ibm/create-cluster-vpc.md

bx2.4x16     4       16GB     8Gbps           UBUNTU_18_64   virtual       100GB     0B                  vpc-gen2   
...
```

The recommended configuration for a cluster is at least 8 vCPU cores with 16GB memory. Hence, we recommend
`bx2.4x16` flavor to create a two-worker-node cluster. Keep in mind that you can always scale the cluster
by adding more worker nodes should your application scales up.

Now set the environment variable with the flavor you choose.

```shell
export WORKER_NODE_FLAVOR=bx2.4x16
```

## Create an IBM Cloud Kubernetes cluster for `vpc-gen2` infrastructure

Creating a `vpc-gen2` based cluster needs a VPC, a subnet and a public gateway attached to it. Fortunately, this is a one
time setup. Future `vpc-gen2` clusters can reuse the same VPC/subnet(with attached public-gateway).

1. Begin with installing a vpc-infrastructure plugin:

    ```shell
    ibmcloud plugin install vpc-infrastructure
    ```
   
   Refer to this [link](https://cloud.ibm.com/docs/containers?topic=containers-vpc_ks_tutorial), for more information.

2. Target the gen 2 to access gen 2 resources:

    ```shell
    ibmcloud is target --gen 2
    ```
   

>    Verify, the target is correctly setup

Reviewer: nit, grammar:

```suggestion
   Verify that the target is correctly set up:
```

>    ```shell
>     ibmcloud is target
>     ```

Reviewer: nit

```suggestion
    ```shell
    ibmcloud is target
    ```
```

>     Example output

Reviewer: ```suggestion
    Example output:
```

    
    ```
    Target Generation: 2
    ```

3. Create or use an existing VPC:
   
   a) Use an existing VPC: 
   
    ```shell
    ibmcloud is vpcs
    ```

    Example output:
    ```
    Listing vpcs for generation 2 compute in all resource groups and region ...
    ID                                          Name                Status      Classic access   Default network ACL                                    Default security group                                 Resource group   
    r006-hidden-68cc-4d40-xxxx-4319fa3gxxxx   my-vpc1              available   false            husker-sloping-bee-resize                              blimp-hasty-unaware-overflow                           kubeflow   
    ```

>     If the above list contains the VPC that can be used to deploy your cluster - note it's ID.

Reviewer: ```suggestion

    If the above list contains the VPC that can be used to deploy your cluster - make a note of its ID.
```

its <- it's

    --zone=$CLUSTER_ZONE \
    --version=${KUBERNETES_VERSION} \
    --flavor ${WORKER_NODE_FLAVOR} \
    --vpc-id ${VPC_ID} \
    --subnet-id ${SUBNET_ID} \
    --workers=2
    ```

6. Attach a public gateway

   This step is mandatory for Kubeflow deployment to succeed, because pods need public internet access to download images.
   
   First check, if your cluster is already assigned a public gateway:
   
    ```shell
    ibmcloud is pubgws
    ```

    Example output:

    ```
    Listing public gateways for generation 2 compute in all resource groups and region ...
    ID                                          Name                                       Status      Floating IP      VPC                 Zone         Resource group   
    r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx   my-gateway                                 available   xxx.xxx.xxx.xxx       my-vpc              us-south-3   default   
   
    ```
   
   In the above run, gateway is already attached for the vpc: `my-vpc`. In case no gateway is attached, proceed with
   rest of the setup.

   
   Attach a public gateway:
   
   ```shell
   ibmcloud is public-gateway-create my-gateway $VPC_ID us-south-3
   ```
   

>    Example output:

Reviewer: ```suggestion

   Example output:
```

   ```
   ID: r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx
   ```

>    Save the above generated gateway ID as follows:

Reviewer: ```suggestion

   Save the above generated gateway ID as follows:
```

   
   ```shell
   export GATEWAY_ID="r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx"
   ```
   

>    Finally, attach the public gateway to the subnet.
>    
>    ```shell
>    ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID
>    ```
>    Example output:
>    ```

Reviewer: ```suggestion
   Finally, attach the public gateway to the subnet:
   
   ```shell
   ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID
   ```

   Example output:

   ```
```

   Updating subnet 0737-27299d09-1d95-4a9d-a491-a6949axxxxxx under account IBM as user new@user-email.com...
                          
   ID                  0737-27299d09-1d95-4a9d-a491-a6949axxxxxx   
   Name                my-subnet   
   ...
   ```

### Verifying the cluster

> To use the created cluster, switch the Kubernetes context to point to the cluster with the command

Reviewer: ```suggestion
To use the created cluster, switch the Kubernetes context to point to the cluster with the followin command:
```

```shell
ibmcloud ks cluster config --cluster ${CLUSTER_NAME}
```

> Make sure all worker nodes are up with the command below

Reviewer: ```suggestion
Make sure all worker nodes are up with the command below:
```

+++
title = "Create or access an IBM Cloud Kubernetes cluster on a VPC"
description = "Instructions for creating or connecting to a Kubernetes cluster on IBM Cloud vpc-gen2"
weight = 4
+++

## Create and setup a new cluster

Follow these steps to create and setup a new IBM Cloud Kubernetes Service(IKS) cluster on `vpc-gen2` provider.

A `vpc-gen2` cluster does not expose each node to the public internet directly and thus has more secure
and more complex network setup. It is recommended setup for secured production use cases of Kubeflow.

### Setting environment variables

Choose the region and the worker node provider for your cluster, and set the environment variables.

```shell
export KUBERNERTES_VERSION=1.18
export CLUSTER_ZONE=us-south-3
export CLUSTER_NAME=kubeflow-vpc
```

where:

> - `KUBERNETES_VERSION` Run `ibmcloud ks versions` to see the supported Kubernetes versions. Refer to

Reviewer: ```suggestion
- `KUBERNETES_VERSION`: Run `ibmcloud ks versions` to see the supported Kubernetes versions. Refer to
```

by adding more worker nodes should your application scales up.

Now set the environment variable with the flavor you choose.

```shell
export WORKER_NODE_FLAVOR=bx2.4x16
```

## Create an IBM Cloud Kubernetes cluster for `vpc-gen2` infrastructure

Creating a `vpc-gen2` based cluster needs a VPC, a subnet and a public gateway attached to it. Fortunately, this is a one
time setup. Future `vpc-gen2` clusters can reuse the same VPC/subnet(with attached public-gateway).

1. Begin with installing a `vpc-infrastructure` plugin:

    ```shell
    ibmcloud plugin install vpc-infrastructure
    ```
   
   Refer to this [link](https://cloud.ibm.com/docs/containers?topic=containers-vpc_ks_tutorial), for more information.

2. Target `vpc-gen 2` to access gen 2 VPC resources:

    ```shell
    ibmcloud is target --gen 2
    ```
   
   Verify that the target is correctly set up:

    ```shell
    ibmcloud is target
    ```

    Example output:
    
    ```
    Target Generation: 2
    ```

3. Create or use an existing VPC:
   
   a) Use an existing VPC: 
   
    ```shell
    ibmcloud is vpcs
    ```

>     Example output:

Reviewer: ```suggestion

    Example output:
```

    Name                                           my-vpc   
    ...  
    ```

   **Save the ID in a variable `VPC_ID` as follows, so that we can use it later.**
   
    ```shell
    export VPC_ID=r006-hidden-68cc-4d40-xxxx-4319fa3fxxxx
    ```

4. Create or use an existing subnet:

    a) To use an existing subnet:
    
    ```shell
    ibmcloud is subnets
    ```

    Example output:
    
    ```
    Listing subnets for generation 2 compute in all resource groups and region ...
    ID                                          Name                      Status      Subnet CIDR       Addresses     ACL                                                    Public Gateway                             VPC                 Zone         Resource group   
    0737-27299d09-1d95-4a9d-a491-a6949axxxxxx   my-subnet                 available   10.240.128.0/18   16373/16384   husker-sloping-bee-resize                              my-gateway                                 my-vpc              us-south-3   kubeflow   
    ```

    If the above list contains the subnet corresponding to your VPC, that can be used to deploy your cluster - make sure
   you note it's ID.
   
    b) To create a new subnet:

>         - List address prefixes and note the CIDR block corresponding to a Zone;

Reviewer: ```suggestion
      - List address prefixes and note the CIDR block corresponding to a Zone;
```

    ID                                          Name                                CIDR block        Zone         Has subnets   Is default   Created   
    r006-xxxxxxxx-4002-46d2-8a4f-f69e7ba3xxxx   rising-rectified-much-brew          10.240.0.0/18     us-south-1   false         true         2021-03-05T14:58:39+05:30   
    r006-xxxxxxxx-dca9-4321-bb6c-960c4424xxxx   retrial-reversal-pelican-cavalier   10.240.64.0/18    us-south-2   false         true         2021-03-05T14:58:39+05:30   
    r006-xxxxxxxx-7352-4a46-bfb1-fcbac6cbxxxx   subfloor-certainly-herbal-ajar      10.240.128.0/18   us-south-3   false         true         2021-03-05T14:58:39+05:30  
    ```

    - Now create a subnet as follows:

    ```shell
    ibmcloud is subnet-create my-subnet $VPC_ID $CLUSTER_ZONE --ipv4-cidr-block "10.240.128.0/18"
    ```

    Example output:
    
    ```
    Creating subnet my-subnet in resource group kubeflow under account IBM as user new@user-email.com...
                          
    ID                  0737-27299d09-1d95-4a9d-a491-a6949axxxxxx   
    Name                my-subnet
    ```

>    - Make sure you export the subnet IDs follows:

Reviewer: ```suggestion
    - Make sure you export the subnet IDs follows:
```

    ```shell
    export SUBNET_ID=0737-27299d09-1d95-4a9d-a491-a6949axxxxxx
    ```

5. Create a `vpc-gen2` based Kubernetes cluster:
    
    ```shell
    ibmcloud ks cluster create vpc-gen2 \
    --name $CLUSTER_NAME \
    --zone $CLUSTER_ZONE \
    --version ${KUBERNETES_VERSION} \
    --flavor ${WORKER_NODE_FLAVOR} \
    --vpc-id ${VPC_ID} \
    --subnet-id ${SUBNET_ID} \
    --workers 2
    ```

6. Attach a public gateway

   This step is mandatory for Kubeflow deployment to succeed, because pods need public internet access to download images.
   

>    First check, if your cluster is already assigned a public gateway:

Reviewer: ```suggestion
    - First, check if your cluster is already assigned a public gateway:
```

(You have sequential steps under "6. Attach a public gateway", so I'm adding "-" to help follow them)

   
    ```shell
    ibmcloud is pubgws
    ```

    Example output:

    ```
    Listing public gateways for generation 2 compute in all resource groups and region ...
    ID                                          Name                                       Status      Floating IP      VPC                 Zone         Resource group   
    r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx   my-gateway                                 available   xxx.xxx.xxx.xxx       my-vpc              us-south-3   default   
   
    ```
   

>    In the above run, gateway is already attached for the vpc: `my-vpc`. In case no gateway is attached, proceed with

Reviewer: ```suggestion
    In the above run, the gateway is already attached for the vpc: `my-vpc`. In case no gateway is attached, proceed with
```

>    rest of the setup.

Reviewer: ```suggestion
    the rest of the setup.
```

>    
>    Attach a public gateway:

Reviewer: ```suggestion
    - Next, attach a public gateway by running the following command:
```

(since you have "6. Attach a public gateway" as the title of the section)

>    ```shell
>    ibmcloud is public-gateway-create my-gateway $VPC_ID $CLUSTER_ZONE
>    ```

Reviewer: ```suggestion
    ```shell
    ibmcloud is public-gateway-create my-gateway $VPC_ID $CLUSTER_ZONE
    ```
```

   Example output:

>    ```
>    ID: r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx
>    ```

Reviewer: ```suggestion
    ```
    ID: r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx
    ```
```

   Save the above generated gateway ID as follows:

>    ```shell
>    export GATEWAY_ID="r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx"
>    ```

Reviewer: ```suggestion
    ```shell
    export GATEWAY_ID="r006-xxxxxxxx-5731-4ffe-bc51-1d9e5fxxxxxx"
    ```
```

>    Finally, attach the public gateway to the subnet:

Reviewer: ```suggestion
    - Finally, attach the public gateway to the subnet:
```

(You have sequential steps under "6. Attach a public gateway", so I'm adding "-" to help follow them)

>    ```
>    ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID
>    Updating subnet 0737-27299d09-1d95-4a9d-a491-a6949axxxxxx under account IBM as user new@user-email.com...
>                           
>    ID                  0737-27299d09-1d95-4a9d-a491-a6949axxxxxx   
>    Name                my-subnet   
>    ...
>    ```

Reviewer: ```suggestion
    ```shell
    ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID
    ```

    Example output:

    ```
    Updating subnet 0737-27299d09-1d95-4a9d-a491-a6949axxxxxx under account IBM as user new@user-email.com...
                          
    ID                  0737-27299d09-1d95-4a9d-a491-a6949axxxxxx   
    Name                my-subnet   
    ...
    ```
```

Reviewer: (I think 4 spaces or 2 tabs (with 2 spaces each) will render better than 3 -- applied across the example, please double check)

### Verifying the cluster

> To use the created cluster, switch the Kubernetes context to point to the cluster with the command

Reviewer: ```suggestion
To use the created cluster, switch the Kubernetes context to point to the cluster:
```

```shell
ibmcloud ks cluster config --cluster ${CLUSTER_NAME}
```

Make sure all worker nodes are up with the command below:

```shell
kubectl get nodes
```

> and make sure all the nodes are in `Ready` state.

Reviewer: ```suggestion
and verify that all the nodes are in `Ready` state.
```

(repetition of "Make sure... make sure...")

   Finally, attach the public gateway to the subnet:
   
   ```

>    ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID

Reviewer: Ideally, this should be consistent with the rest of the guides, so:

```
    ```shell
    ibmcloud is subnet-update $SUBNET_ID --public-gateway-id $GATEWAY_ID
    ````

    Example output:

    ```
    Updating subnet 0737-27299d09-1d95-4a9d-a491-a6949axxxxxx under account IBM as user new@user-email.com...
    ...
    ...
    ```
```

(Without the indentation)

Author: Hi @8bitmp3, thank you for the careful examination. 

Indentation here has a purpose to nest items according the bullet points. Without it, it will not render properly.
